controllers/internalrequest/adapter.go

```diff
@@ -18,34 +18,134 @@ package internalrequest
 
 import (
 	"context"
+	"fmt"
 	"github.com/go-logr/logr"
+	libhandler "github.com/operator-framework/operator-lib/handler"
 	"github.com/redhat-appstudio/internal-services/api/v1alpha1"
+	"github.com/redhat-appstudio/internal-services/loader"
+	"github.com/redhat-appstudio/internal-services/tekton"
 	"github.com/redhat-appstudio/operator-goodies/reconciler"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	"knative.dev/pkg/apis"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
 )
 
 // Adapter holds the objects needed to reconcile a Release.
 type Adapter struct {
 	client          client.Client
-	context         context.Context
+	ctx             context.Context
 	internalClient  client.Client
 	internalRequest *v1alpha1.InternalRequest
+	loader          loader.ObjectLoader
 	logger          logr.Logger
 }
 
 // NewAdapter creates and returns an Adapter instance.
-func NewAdapter(internalRequest *v1alpha1.InternalRequest, client, internalClient client.Client, context context.Context, logger logr.Logger) *Adapter {
+func NewAdapter(ctx context.Context, client, internalClient client.Client, internalRequest *v1alpha1.InternalRequest, loader loader.ObjectLoader, logger logr.Logger) *Adapter {
 	return &Adapter{
 		client:          client,
-		context:         context,
+		ctx:             ctx,
 		internalRequest: internalRequest,
 		internalClient:  internalClient,
+		loader:          loader,
 		logger:          logger,
 	}
 }
 
-func (a *Adapter) EnsureReconcileOperationIsLogged() (reconciler.OperationResult, error) {
-	a.logger.Info("InternalRequest successfully watched")
+// EnsureRequestIsHandled is an operation that will ensure that the InternalRequest is handled by creating a new
+// PipelineRun for the Pipeline referenced in the Request field.
+func (a *Adapter) EnsureRequestIsHandled() (reconciler.OperationResult, error) {
```

Comment:
complete nit pick and I will leave it up to you but I'd prefer a name like `EnsurePipelineRunCreatedForRequest` instead of having to figure out what "handling" one means, but I am sure you have your reasons

Reply:
I'm ok with it, but shortening it to `EnsurePipelineRunIsCreated`.